src/functions/severa/get-filtered-workhours/handler.ts

```diff
@@ -11,56 +11,64 @@ import type SeveraResponseWorkHours from "src/types/severa/workHour/severaRespon
+    if (severaProjectId) {
+      // Request in parallel for each opted-in user
+      const workHoursResults = await Promise.all(
+        optInUsers.map(async (user) => {
+          try {
+            const url = new URL(`${process.env.SEVERA_DEMO_BASE_URL}/v1/users/${user.guid}/workhours`);
+            // This endpoint/query parameter is working but it is not documented proper in Severa API spec
+            url.searchParams.append("projectGuid", severaProjectId);
+            if (startDate) url.searchParams.append("startDate", startDate);
+            if (endDate) url.searchParams.append("endDate", endDate);
+            const userWorkHours = await api.getWorkHours(url);
+            return userWorkHours;
+          } catch (e) {
+            errors.push({ userGuid: user.guid, error: e });
+            return [];
+          }
+        })
+      );
+      // Combine all results
+      allWorkHours = workHoursResults.flat();
+      // Filtering only for the specified project
+      allWorkHours = allWorkHours.filter(
+        (workHours) => workHours.project?.guid == severaProjectId
+      );
+    } else {
+      // Standard logic for other cases (by userId or without projectId)
+      // Manually build the URL since buildWorkHoursUrl does not exist
+      let url: URL;
+      if (severaUserId) {
+        url = new URL(`${process.env.SEVERA_DEMO_BASE_URL}/v1/users/${severaUserId}/workhours`);
       } else {
-        endpointPath = "workhours";
+        throw new Error("severaUserId is required when severaProjectId is not specified");
       }
+      if (startDate) url.searchParams.append("startDate", startDate);
+      if (endDate) url.searchParams.append("endDate", endDate);
 
-      const customUrl = new URL(`${process.env.SEVERA_DEMO_BASE_URL}/v1/${endpointPath}`);
-
-      if (startDate) customUrl.searchParams.append("startDate", startDate);
-      if (endDate) customUrl.searchParams.append("endDate", endDate);
+      allWorkHours = await api.getWorkHours(url);
+    }
 
-      return customUrl;
-    };
-
-    const url = buildWorkHoursUrl(severaProjectId, severaUserId, startDate, endDate);
-    const response = await api.getWorkHours(url);
-
-    const filteredWorkHours = response.filter((workHours: SeveraResponseWorkHours) => {
-    /**
-     * Note: If severaProjectId and severaUserId are both true, the work hours data
-     * will be filtered by project in the Severa API call due to workHours custom url.
-     */
-      if (severaProjectId && severaUserId) {
-        const filteredWorkHoursUserProject = FilterUtilities.filterByUserSevera(workHours.user?.guid, severaUserId);
-        if (!filteredWorkHoursUserProject) return false ;
+    // Filter by opted-in users (just in case there are any "unauthorized" users in the response)
+    const filteredWorkHours = allWorkHours.filter((workHours: SeveraResponseWorkHours) => {
+      if (!workHours.user?.guid || !optInUserGuids.has(workHours.user.guid)) {
+        return false;
       }
```

Comment:
This is a good idea, but it should be done before the request is made, then we can be sure we are not fetching any non optIn user data.